src/main/java/seedu/address/model/person/predicates/ProductContainsKeywordPredicate.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.model.person.predicates;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Product} matches any of the keywords given.
+ */
+public class ProductContainsKeywordPredicate extends SupplierPredicate {
+
+    public ProductContainsKeywordPredicate(String keyword) {
+        super(keyword);
+    }
+
+    @Override
+    public boolean test(Person person) {
+        // Partial String matching unlike the normal AB3
+        return person.getProducts().stream()
+                        .anyMatch(product -> product.toString().toLowerCase().contains(keyword.toLowerCase()));
```

Comment:
While I understand partial string matching will definitely be beneficial for finding supplier by name and company name, do you think it would be better to not have partial string matching for finding supplier by product? I think we should discuss this so that we can standardise with the find command for deliveries.